### PR TITLE
Loss

### PR DESCRIPTION
**Pull request by eellison**
Adding Loss modules to script.  Some of the modules have an optional tensor parameter. I will wait until wanchao's diff to support optional tensors is landed before landing this.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -10698,6 +10711,145 @@ def wait_script(x1, x2, x3):
         self.assertEqual(y2, foo2(x1, x2))
         self.assertEqual(y3, foo3(x1, x2, x3))
 
+
```
Comment (driazati) [inline test/test_jit.py]: These can go into `additional_module_tests` to keep everything extra in 1 place

File: test/test_jit.py [hunk 2]
```diff
@@ -10698,6 +10711,145 @@ def wait_script(x1, x2, x3):
...
+loss_tests = [  # noqa: C408
```
Comment (wanchaol) [inline test/test_jit.py]: Couldn't we just use tests from common_nn instead of adding them here?

Comment (eellison) [reply test/test_jit.py]: Requires a fair amount of refactoring, can do it in a future diff

**Review by wanchaol — APPROVED** [review]
LG as long as CI passes

**Comment by wanchaol** [comment]
Hmm apparently there're several tests are failing (BCELoss, MarginRankingLoss, etc.) can you take a look?

**Pull request closed without merging**